tests/containers/secret.pm

```diff
@@ -0,0 +1,89 @@
+# SUSE's openQA tests
+#
+# Copyright 2023-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman, docker
+# Summary: Test the `secret` subcommand for Docker and Podman
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use containers::common;
+use containers::container_images;
+
+sub run {
+    my ($self, $args) = @_;
+    my $runtime = $args->{runtime};
+    my $output = '';
+
+    select_serial_terminal();
+
+    # In Docker, secrets can only be used in a swarm
+    if ($runtime =~ 'docker') {
+        my $ip_addr = script_output(qq(ip -6 address show dev eth0 |
+        awk '/inet6/{split(\$2, a, "/"); print a[1]; exit;}'));
+        record_info("docker swarm init", "Initializing docker swarm in IP $ip_addr");
+        assert_script_run("docker swarm init --advertise-addr $ip_addr");
+    }
+
+    # Create a secret1 from file and inspect it
+    record_info("secret create file");
+    script_run("printf T0p_S3cr3t1 > secret1.txt");
+    assert_script_run("$runtime secret create secret1 secret1.txt", fail_message => "Error creating secret from file", timeout => 60);
+    record_info("secret inspect file", script_output("$runtime secret inspect secret1"));
+
+    # Create a secret2 from CLI and inspect it
+    record_info("secret create CLI");
+    assert_script_run("printf T0p_S3cr3t2 | $runtime secret create secret2 -", fail_message => "Error creating secret from CLI", timeout => 60);
+    record_info("secret inspect CLI", script_output("$runtime secret inspect secret2"));
+
+    # Check if secret exists (only in podman)
+    if ($runtime =~ 'podman') {
+        record_info("secret exists");
+        assert_script_run("podman secret exists secret1", fail_message => "Error checking if secret exists");
+        assert_script_run("podman secret exists secret2", fail_message => "Error checking if secret exists");
+        # This secret3 does not exist and thus `secret exists` must return 1
+        assert_script_run("! podman secret exists secret3", fail_message => "Error checking if secret exists");
+    }
+
+    # List all secrets
+    record_info("secret ls");
+    assert_script_run("$runtime secret ls");
+
+    # Run two containers passing secret1 as an env variable and secret2 as default
+    my $runtime_command = ($runtime =~ 'docker') ? 'service create' : 'run';
+    record_info("Access secrets");
+    script_run("$runtime pull registry.opensuse.org/opensuse/bci/bci-busybox:latest", timeout => 120);
+
+    # secret1 testing (default secret)
+    validate_script_output("$runtime $runtime_command --name secret1-test --secret secret1 bci-busybox:latest cat /run/secrets/secret1", sub { m/T0p_S3cr3t1/ });
+    # Commit the container and check that the secrets are not in it
+    record_info("Commit container secret1-test");
```

Comment:
`record_info` method has 2 parameters, the first one is what it's displayed in the openqa box/frame and it shall be short, because long strings are cut and not readable. The second parameter can be as long as you want, as it's visible when you click on the box/frame. So, I would prefer something like:
`record_info("INFO", "Commit container secret1-test")` or `record_info("Commit", "Commit container secret1-test")`